R/clone_package.R

```diff
@@ -1,287 +1,325 @@
+#' package <- arcticdatautils::create_dummy_package(to@@mn)
 #'
+#' # Clone object
+#' cloned_package <- clone_package(resource_map_pid = package$resource_map
+#'                           from = from,
+#'                            to = to,
+#'                            add_access_to = arcticdatautils:::get_token_subject(),
+#'                            public = TRUE,
+#'                            new_pid = TRUE)
 #' }
-#'
 #' @export
 clone_package <- function(resource_map_pid,
                           from,
                           to,
+                          add_access_to,
                           public = FALSE,
-                          clone_child_packages = FALSE) {
-    #' TODO - create dynamic structure that allows for more than one level of
-    #' children (3+ nesting levels)
-    #' TODO - update pids in EML
-    stopifnot(is.character(resource_map_pid))
-    stopifnot(methods::is(from, "MNode"))
-    stopifnot(methods::is(to, "MNode"))
-    stopifnot(is.logical(public))
-    stopifnot(is.logical(clone_child_packages))
-
-    # Get package information
-    package <- arcticdatautils::get_package(from, resource_map_pid)
-
-    # Clone the parent package
-    response <- clone_one_package(package$resource_map, from, to)
-
-    if (clone_child_packages == TRUE) {
-        n_child_packages <- length(package$child_packages)
-
-        if (n_child_packages > 0) {
-            progressBar <- utils::txtProgressBar(min = 0, max = n_child_packages, style = 3)
-
-            # Clone child packages
-            child_packages <- unlist(lapply(seq_len(n_child_packages), function(i) {
-                pids <- clone_one_package(package$child_packages[i], from, to)
-                utils::setTxtProgressBar(progressBar, i)
-                pids
-            }))
-
-            # Select resource_map_pid(s) of child packages
-            indices <- which(names(child_packages) == "resource_map")
-            child_resource_map_pids <- child_packages[indices]
-
-            # Nest child packages
-            updated_resource_map_pid <- arcticdatautils::update_resource_map(to,
-                                                                             response$resource_map,
-                                                                             response$metadata,
-                                                                             response$data,
-                                                                             child_resource_map_pids)
-
-            response[["resource_map"]] <- updated_resource_map_pid
-            response[["child_packages"]] <- unname(child_resource_map_pids)
+                          clone_children = FALSE,
+                          new_pid = TRUE,
+                          new_submitter = NULL,
+                          change_origin_node = FALSE,
+                          change_auth_node = FALSE) {
+
+    if (!(is.character(resource_map_pid) & nchar(resource_map_pid) > 0)) {
```

Comment:
These function checks could probably be condensed, perhaps by using something like `assertthat`, which does some of the error-message to English translation for you: https://github.com/hadley/assertthat

Reply:
Good tip, will keep for future reference but don't feel is necessary to go back through these as the changes will be purely aesthetic.